frontend/lib/resolution/Resolver.cpp

```diff
@@ -1556,16 +1559,19 @@ void Resolver::resolveNamedDecl(const NamedDecl* decl, const Type* useType) {
   // adjust tuple declarations for value / referential tuples
   if (typePtr != nullptr && decl->isVarArgFormal() == false) {
     if (auto tupleType = typePtr->toTupleType()) {
-      if (declaredKind == QualifiedType::DEFAULT_INTENT ||
-          declaredKind == QualifiedType::CONST_INTENT) {
+      if (declaredKind == QualifiedType::DEFAULT_INTENT) {
         typePtr = tupleType->toReferentialTuple(context);
         qtKind = QualifiedType::CONST_REF;
+      } else if (declaredKind == QualifiedType::CONST_INTENT) {
+        typePtr = tupleType->toReferentialTuple(context, true);
```

Comment:
Here and elsewhere,

```suggestion
        typePtr = tupleType->toReferentialTuple(context, /* makeConst */ true);
```

To make it clearer what's going on.